### PR TITLE
Cross version testing support (part 2)

### PR DESCRIPTION
**Pull request by aslesarenko**
This PR extends cross version testing for Sigma operations.

### CODE REVIEW

File: sigmastate/src/main/scala/sigmastate/eval/CostingDataContext.scala [hunk 1]
```diff
@@ -89,11 +89,9 @@ case class CAvlTree(treeData: AvlTreeData) extends AvlTree with WrapperOf[AvlTre
 
   override def wrappedValue: AvlTreeData = treeData
 
-  def startingDigest: Coll[Byte] = Colls.fromArray(treeData.digest)
+  override def keyLength: Int = treeData.keyLength
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/eval/CostingDataContext.scala:92]: Please provide ScalaDoc for all the public methods being touched in CAvlTree as well as for the class itself.

File: sigmastate/src/main/scala/sigmastate/lang/Terms.scala [hunk 2]
```diff
@@ -173,6 +174,13 @@ object Terms {
   }
   object MethodCall extends ValueCompanion {
     override def opCode: OpCode = OpCodes.MethodCallCode
+    def typed[T <: SValue](obj: Value[SType],
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/lang/Terms.scala:177]: Please provide ScalaDoc.

File: sigmastate/src/main/scala/sigmastate/types.scala [hunk 3]
```diff
@@ -1294,6 +1294,14 @@ object STuple extends STypeCompanion {
     }
 }
 
+object SPair {
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/types.scala:1297]: ScalaDoc

File: sigmastate/src/test/scala/special/sigma/SigmaDslSpec.scala [hunk 4]
```diff
@@ -5,24 +5,27 @@ import java.math.BigInteger
 import org.ergoplatform.ErgoScriptPredef.TrueProp
 import org.ergoplatform.dsl.{SigmaContractSyntax, TestContractSpec}
```
Comment (kushti) [inline sigmastate/src/test/scala/special/sigma/SigmaDslSpec.scala:6]: Unused import (and not only here in this file)

File: sigmastate/src/test/scala/special/sigma/SigmaDslSpec.scala [hunk 5]
```diff
@@ -973,444 +1681,1018 @@ class SigmaDslSpec extends SigmaDslTesting { suite =>
...
+              SelectField.typed[Value[SInt.type]](ValUse(4, SPair(SInt, SInt)), 2.toByte)
+            ),
+            Map()
+          )
+        )
+      ))
+
+    forAll(collWithRangeGen) { data =>
       val arr = data._1
       val range = data._2
```
[the remark this answers is not in the document]
Comment (aslesarenko) [reply sigmastate/src/test/scala/special/sigma/SigmaDslSpec.scala:2238]: fixed

File: sigmastate/src/main/scala/sigmastate/eval/CostingDataContext.scala (hunk 1 above)
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/eval/CostingDataContext.scala:92]: done.

File: sigmastate/src/main/scala/sigmastate/lang/Terms.scala (hunk 2 above)
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/lang/Terms.scala:177]: done

File: sigmastate/src/main/scala/sigmastate/types.scala (hunk 3 above)
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/types.scala:1297]: done

File: sigmastate/src/test/scala/special/sigma/SigmaDslSpec.scala (hunk 4 above)
Comment (aslesarenko) [reply sigmastate/src/test/scala/special/sigma/SigmaDslSpec.scala:6]: done

**Comment by kushti** [comment]
On  inserts and updates into AVL+  tree, not bug at all. Please note that a proof is an object which allows the Verifier to get a new digest. In your case the Verifier is able to do that, the digest would be different from expected though. See this new test https://github.com/ScorexFoundation/sigmastate-interpreter/pull/674 . In contracts, the digest is checked like "sigmaProp(tree.insert(ops, proof).get == endTree)"
More on the Verifier and the proof can be found at https://www.youtube.com/watch?v=PHY7JnLrK5o

**Pull request merged**